### PR TITLE
Implement Dataframe.iterrows using spark.dataframe.toLocalIterator

### PR DESCRIPTION
**Pull request by guyao**
#885 
Inspired by https://github.com/databricks/koalas/issues/1003#issuecomment-556908359

the data type yield from iterator is `pd.Series`. I assume each row is small enough.

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -1100,6 +1100,56 @@ def iteritems(self) -> Iterable:
         cols = list(self.columns)
         return list((col_name, self[col_name]) for col_name in cols)
 
+    def iterrows(self):
+        """
+        Iterate over DataFrame rows as (index, Series) pairs.
+        Yields
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: Looks you should add some spaces between the doc and `Yields`.

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -1100,6 +1100,56 @@ def iteritems(self) -> Iterable:
...
+            The data of the row as a Series.
+        it : generator
+            A generator that iterates over the rows of the frame.
+
+        Notes
+        -----
+        1. Because ``iterrows`` returns a Series for each row,
+           it does **not** preserve dtypes across the rows (dtypes are
+           preserved across columns for DataFrames). For example,
+           >>> df = pd.DataFrame([[1, 1.5]], columns=['int', 'float'])
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: here too

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -1100,6 +1100,56 @@ def iteritems(self) -> Iterable:
...
+           float    1.5
+           Name: 0, dtype: float64
+           >>> print(row['int'].dtype)
+           float64
+           >>> print(df['int'].dtype)
+           int64
+           To preserve dtypes while iterating over the rows, it is better
+           to use :meth:`itertuples` which returns namedtuples of the values
+           and which is generally faster than ``iterrows``.
+        2. You should **never modify** something you are iterating over.
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: and here too

File: databricks/koalas/frame.py (hunk 2 above)
Comment (ueshin) [reply databricks/koalas/frame.py]: This is not addressed yet.

Comment (guyao) [reply databricks/koalas/frame.py]: Thanks, fixed!

File: databricks/koalas/frame.py [hunk 4]
```diff
@@ -1100,6 +1100,59 @@ def iteritems(self) -> Iterable:
...
+           >>> row = next(df.iterrows())[1]
+           >>> row
+           int      1.0
+           float    1.5
+           Name: 0, dtype: float64
+           >>> print(row['int'].dtype)
+           float64
+           >>> print(df['int'].dtype)
+           int64
+           To preserve dtypes while iterating over the rows, it is better
```
[the remark this answers is not in the document]
Comment (guyao) [reply databricks/koalas/frame.py]: Thanks, fixed! 
Added multi index test in `test/test_dataframe`

File: databricks/koalas/frame.py [hunk 5]
```diff
@@ -1110,6 +1110,7 @@ def iterrows(self):
             The index of the row. A tuple for a `MultiIndex`.
         data : pandas.Series
             The data of the row as a Series.
+
```
Comment (ueshin) [inline databricks/koalas/frame.py]: nit: should revert this change?

Comment (guyao) [reply databricks/koalas/frame.py]: I saw an empty line between `data` and `it` in pandas doc 
https://github.com/pandas-dev/pandas/blob/v0.25.3/pandas/core/frame.py#L860-L910

This function returns a generator `it` and items are `index` and `data`. That might be the reason an empty line here

Comment (ueshin) [reply databricks/koalas/frame.py]: I see, then maybe we should add `Returns` section?
```
        Returns
        -------
        it : generator
            ...
```

Comment (guyao) [reply databricks/koalas/frame.py]: Fixed!

Comment (ueshin) [reply databricks/koalas/frame.py]: ah, sorry, seems like we can't use both `Yields` and `Returns` at the same time.

```
ValueError: Docstring contains both a Returns and Yields section.
```

Could you revert the last commit?

Comment (guyao) [reply databricks/koalas/frame.py]: oh interesting. 
sorry I should try to build the doc to test
branch updated

**Comment by ueshin** [comment]
Thanks! merging.
